oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/IndexName.java

```diff
@@ -231,26 +231,37 @@ public String nextCustomizedName() {
     }
 
     public static boolean isIndexActive(String indexPath, NodeState rootState) {
+        return isIndexActive(indexPath, rootState, 0);
+    }
+
+    private static boolean isIndexActive(String indexPath, NodeState rootState, int recursionDepth) {
         // An index is active if it has a hidden child node that starts with ":oak:mount-",
         // OR if it is an active merged index
-        try {
-            NodeState indexNode = rootState;
-            for (String e : PathUtils.elements(indexPath)) {
-                indexNode = indexNode.getChildNode(e);
+        if (recursionDepth > 50) {
+            LOG.warn("Fail to check index activeness for {} due to high recursion depth", indexPath);
+            return true;
+        }
+        NodeState indexNode = rootState;
+        for (String e : PathUtils.elements(indexPath)) {
+            indexNode = indexNode.getChildNode(e);
+        }
+        for (String c : indexNode.getChildNodeNames()) {
+            if (c.startsWith(":oak:mount-")) {
+                return true;
             }
-            for (String c : indexNode.getChildNodeNames()) {
-                if (c.startsWith(":oak:mount-")) {
-                    return true;
-                }
+        }
+        if (recursionDepth >= 2) {
```

Comment:
Shouldn't it >= 1? The recursion starts at 0, so at level 1 we can already have an OOTB index, no? Is there a test for this case? That is, for an index `lucene-7-custom-1` with `merges=["lucene-7", "lucene-6-custom2"]`. And are there tests for other cases? I don't see them in this PR or in the IndexNameTest class.

Reply:
Level 1 (the direct children) will still need to check. 

> Is there a test for this case?

Yes: CustomizedIndexTest and IndexNameTest are in the patch. The test coverage of the changes is 100% as far as I see.